packages/babel-plugin-transform-bigint/src/index.js

```diff
@@ -0,0 +1,41 @@
+import syntaxBigInt from "babel-plugin-syntax-bigint";
+
+export default function({ types: t }) {
+  return {
+    inherits: syntaxBigInt,
+
+    visitor: {
+      Program: {
+        exit(path: NodePath, state: Object) {
+          state.file.addImport("bigint-polyfill", "default", "BigInt");
+        },
+      },
+      BinaryExpression(path) {
+        if (path.node.operator === "+") {
+          if (
+            path.node.left.type === "BigIntLiteral" &&
+            path.node.left.type === "BigIntLiteral"
+          ) {
```

Comment:
Just complementing @littledan, we probably don't have ```path.node.*.type``` when one of the operands isn't a literal, so we need to have a check at runtime in all BigInt compatible BinaryExpressions if the kind of this operand is a ```BigInt``` or not.
Also IMHO, expanding into function calls will make our life easier to follow the spec and implement runtime errors due to invalid operand, such as ```10n + 1```.